lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/Resources.kt

```diff
@@ -0,0 +1,98 @@
+@file:JvmName("Resources")
+
+package com.swmansion.starknet.data.types.transactions
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@Serializable
+sealed class Resources {
+    abstract val steps: Int
+    abstract val memoryHoles: Int?
+    abstract val rangeCheckApplications: Int?
+    abstract val pedersenApplications: Int?
+    abstract val poseidonApplications: Int?
+    abstract val ecOpApplications: Int?
+    abstract val ecdsaApplications: Int?
+    abstract val bitwiseApplications: Int?
+    abstract val keccakApplications: Int?
+    abstract val segmentArenaApplications: Int?
+}
+
+@Serializable
+data class ComputationResources(
+    @SerialName("steps")
+    override val steps: Int,
```

Comment:
technically this cannot be 0. do you think we should make a serializer that ensures that, or do we yolo it and can just depend on the node's response?

Reply:
I thought about that and it seems to me yolo is the best approach right now 😅 

We already go in the direction of putting less restrictions than in the spec: `Felt` instead of `ContractAddress`, non-optional fields because of the nodes' deviations, allowing `ingoreUnknownKeys` etc. 

I suppose we can put faith into nodes being correct, and if they aren't, at least entire operation won't fail because of an incorrect zero.